plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -202,7 +202,7 @@ private void evaluateFacetConfiguration() {
   private void updatePageBasedOnFacetConfig(boolean hasCorrectFacetConfiguration) {
     if (hasCorrectFacetConfiguration) {
       canSetMessage = true;
-      setErrorMessage(null);
+      content.getDataBindingContext().updateTargets();
```

Comment:
To summarize: the issue in https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/880#issuecomment-256149028 was caused by the difference between GTK2 and GTK3, and the eventual removal of `setErrorMessage(null)` and the moving the logic from the paint listener to the `setVisible()` call (to be precise the latter is sufficient) fixed it. (Along with the problem that on GTK2 the error message regarding the invalid facet configuration for the deploy page did not appear (for the same reason: it was set in the paint listener which was not executed on GTK2)).

So I'm pretty confident that the error in https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/1546#discussion_r105149487 is not a regression, and was present before this PR.